client/app/models/milestone.js

```diff
@@ -107,4 +129,88 @@ export default class MilestoneModel extends Model {
     // get date diff by accounting for DST
     return getTimeDiffText(remainingTimeInterval);
   }
+
+  @computed('project.dcpPublicstatus', 'dcpMilestone', 'dcpActualstartdate', 'dcpActualenddate', 'dcpReviewmeetingdate')
+  get displayDate() {
+    if ([
+      BOROUGH_BOARD_REFERRAL,
+      BOROUGH_PRESIDENT_REFERRAL,
+      COMMUNITY_BOARD_REFERRAL,
+      CITY_COUNCIL_REVIEW,
+      CPC_PUBLIC_MEETING_PUBLIC_HEARING,
+      EIS_DRAFT_SCOPE_REVIEW,
+      FEIS_SUBMITTED_AND_REVIEW,
+      MAYORAL_VOTE,
+    ].includes(this.dcpMilestone) && this.project.get('dcpPublicstatus') !== 717170000) {
+      return this.dcpActualstartdate;
+    }
+
+    if (this.dcpMilestone === FILED_EAS_REVIEW) {
+      return this.dcpActualstartdate;
+    }
+
+    if ([
+      FINAL_LETTER_SENT,
+      FINAL_SCOPE_OF_WORK_ISSUED,
+      CPC_PUBLIC_MEETING_VOTE,
+      DEIS_PUBLIC_HEARING_HELD,
+      NOC_OF_DRAFT_EIS_ISSUED,
+      REVIEW_SESSION_CERTIFIED_REFERRED,
+    ].includes(this.dcpMilestone) && this.project.get('dcpPublicstatus') !== 717170000) {
+      return this.dcpActualenddate;
+    }
+
+    if ([
+      EIS_PUBLIC_SCOPING_MEETING,
+      PREPARE_FILED_LAND_USE_APPLICATION,
+      LAND_USE_FEE_PAYMENT,
+      CEQR_FEE_PAYMENT,
+      CPC_REVIEW_OF_COUNCIL_MODIFICATION,
+      DEIS_SCOPE_OF_WORK_RELEASED,
+      SCOPING_MEETING,
+    ].includes(this.dcpMilestone)) {
+      return this.dcpActualenddate;
+    }
+
+    /*
+    * The following is for task 13239
+    * The date displayed should be the CRM Review Meeting date for certain milestones
+    * if the milestone statuscode is "in progress" or "completed", and the meeting date is <= current date
+    * I have included a check to make sure that the field is not null to make sure we are not overwriting dates with nothing
+    * Statuscode optionset:
+    *   2 == completed
+    *   717170000 == progress
+    */
+    if ([
+      REVIEW_SESSION_CERTIFIED_REFERRED, // aka 'Application Reviewed at City Planning Commission Review Session'
+      CPC_PUBLIC_MEETING_VOTE, // aka 'City Planning Commission Vote'
+      CPC_PUBLIC_MEETING_PUBLIC_HEARING, // aka 'CPC Public Meeting - Public Hearing' a.k.a 'City Planning Commission Review'
+      REVIEW_SESSION_PRE_HEARING_REVIEW_POST_REFERRAL, // aka 'Review Session - Pre-Hearing Review / Post Referral'
+      REVIEW_SESSION_POST_HEARING_FOLLOW_UP_FUTURE_VOTES, // aka 'Review Session - Post Hearing Follow-Up / Future Votes'
+    ].includes(this.dcpMilestone)) {
+      if ((this.statuscode === 2) || (this.statuscode === 717170000)) {
+        if ((this.dcpReviewmeetingdate) && (Date.now() >= this.dcpReviewmeetingdate.getTime())) {
+          return this.dcpReviewmeetingdate;
+        }
+      }
+    }
+
+    return null;
+  }
```

Comment:
Only alternative approach I can think of is to group things by whether their project `dcpPublicstatus` value is/isn't 717170000. Can you tell me if this computed is "order-sensitive"? For example, if you moved one of these conditional blocks around, would it behave differently? If so, it's sort of an implicit behavior, and could easily confuse someone later.

Reply:
Oh wow, really good question and something I didn't even realize. Here it _is_ order sensitive but that is a mistake/bug during this refactor! In the original transform, each conditional block only assigned to a value, but did not immediately `return`. The last block here was run _in addition_ to previous blocks for certain milestones, in order to override the value assigned by previous blocks. It previously wasn't order sensitive.

However I changed it so each block `return`s, and that makes it order sensitive. Will change it! THANK YOU @allthesignals.